plugins/github-pull-requests-board/src/api/useGetPullRequestsFromUser.ts

```diff
@@ -0,0 +1,110 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+
+import {
+  GraphQlUserPullRequests,
+  PullRequestsNumberAndOwner,
+} from '../utils/types';
+import { useOctokitGraphQl } from './useOctokitGraphQl';
+
+const PULL_REQUEST_LIMIT = 10;
+const GITHUB_GRAPHQL_MAX_ITEMS = 100;
+
+export const useGetPullRequestsFromUser = () => {
+  const graphql =
+    useOctokitGraphQl<GraphQlUserPullRequests<PullRequestsNumberAndOwner[]>>();
+
+  const fn = React.useRef(
+    async (
+      userLogin: string,
+      organization?: string,
+      pullRequestLimit?: number,
+    ): Promise<PullRequestsNumberAndOwner[]> => {
+      const limit = pullRequestLimit ?? PULL_REQUEST_LIMIT;
+
+      return await getPullRequestEdges(graphql, userLogin, limit, organization);
+    },
+  );
+
+  return fn.current;
+};
+
+async function getPullRequestEdges(
+  graphql: (
+    path: string,
+    options?: any,
+  ) => Promise<GraphQlUserPullRequests<PullRequestsNumberAndOwner[]>>,
+  userLogin: string,
+  pullRequestLimit: number,
+  organization?: string,
+): Promise<PullRequestsNumberAndOwner[]> {
+  const pullRequestEdges: PullRequestsNumberAndOwner[] = [];
+  let result:
+    | GraphQlUserPullRequests<PullRequestsNumberAndOwner[]>
+    | undefined = undefined;
+
+  do {
+    result = await graphql(
+      `
+        query ($login: String!, $first: Int, $endCursor: String) {
+          user(login: $login) {
+            pullRequests(states: OPEN, first: $first, after: $endCursor) {
+              edges {
+                node {
+                  number
+                  repository {
+                    name
+                    owner {
+                      login
+                    }
+                  }
+                }
+              }
+              pageInfo {
+                hasNextPage
+                endCursor
+              }
+            }
+          }
+        }
+      `,
+      {
+        login: userLogin,
+        first:
+          pullRequestLimit > GITHUB_GRAPHQL_MAX_ITEMS
+            ? GITHUB_GRAPHQL_MAX_ITEMS
+            : pullRequestLimit,
+        endCursor: result
+          ? result.user.pullRequests.pageInfo.endCursor
+          : undefined,
+      },
+    );
+
+    pullRequestEdges.push(
+      ...result.user.pullRequests.edges.filter(
+        edge =>
+          !organization ||
+          edge?.node?.repository?.owner?.login?.toLocaleLowerCase('en-US') ===
+            organization.toLocaleLowerCase('en-US'),
+      ),
+    );
+
+    if (pullRequestEdges.length >= pullRequestLimit) return pullRequestEdges;
```

Comment:
should you `.slice` it to the right length?

Reply:
This is based on the existing implementation of `useGetPullRequestsFromRepository`. I guess the original intent behind this is to prevent making too many requests towards the Github API, not to return a specific amount of PRs 🤔 

I can add a `.slice` in both `useGetPullRequestsFromRepository` and `useGetPullRequestsFromUser` if you feel it makes more sense though, just let me know what you'd prefer